lib/myxql/connection.ex

```diff
@@ -546,13 +555,35 @@ defmodule MyXQL.Connection do
     rescue
       ArgumentError -> nil
     else
+      # :statement query already prepared
       {^statement, num_params, statement_id, ref} ->
         %{query | num_params: num_params, statement_id: statement_id, ref: ref}
 
       {_statement, _num_params, statement_id, _ref} ->
         Client.com_stmt_close(state.client, statement_id)
         :ets.delete(state.queries, name)
         nil
+
+      # :reference query already prepared
+      {_num_params, statement_id, _ref} ->
+        Client.com_stmt_close(state.client, statement_id)
+        :ets.delete(state.queries, name)
+        nil
+    end
+  end
+
+  defp query_member?(%{queries: nil}, _), do: false
+  defp query_member?(_, %{name: ""}), do: false
+
+  defp query_member?(%{queries: queries}, %Query{name: name, ref: ref}) do
+    try do
+      :ets.lookup_element(queries, name, 2)
+    rescue
+      ArgumentError -> false
+    else
+      {_statement, _num_params, _statement_id, ^ref} -> true
+      {_num_params, _statement_id, ^ref} -> true
+      _ -> false
```

Comment:
How hard would it be to store the `ref` as a separate third element in the tuple and we only look it up instead? The current mechanism means we are copying the statement every time only to check the `ref`.

Reply:
It's not bad. For reference, Postgrex stores `{name, ref, query}` so here could store `{name, ref, {statement, num_params, statement_id, ref}}`.

Now that you mention this, the other lookup can be optimized as well but it's a bit more awkward. I will leave a comment where it is.